src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -374,6 +484,38 @@ public FilterAndTransform(string filterAndPayloadSpec, int startIdx, int endIdx,
                     }
                 }
 
+                Action<string, string, IEnumerable<KeyValuePair<string, string>>> writeEvent = null;
+                if (activityName != null && activityName.Contains("Activity"))
+                {
+                    MethodInfo writeEventMethodInfo = typeof(DiagnosticSourceEventSource).GetTypeInfo().GetDeclaredMethod(activityName);
+                    if (writeEventMethodInfo != null)
+                    {
+                        // This looks up the activityName (which needs to be a name of an event on DiagnosticSourceEventSource
+                        // like Activity1Start and returns that method).   This allows us to have a number of them and this code
+                        // just works.  
+                        try
+                        {
+                            writeEvent = (Action<string, string, IEnumerable<KeyValuePair<string, string>>>)
+                                writeEventMethodInfo.CreateDelegate(typeof(Action<string, string, IEnumerable<KeyValuePair<string, string>>>), _eventSource);
+                        }
+                        catch (Exception) { }
+                    }
+                    if (writeEvent == null)
+                        _eventSource.Message("DiagnosticSource: Could not find Event to log Activity " + activityName);
```

Comment:
It is possible for Events to have messages, and localizing them is reasonable (but still often not done), but event names are not localized (they are treated like API names).